include/gridtools/sid/simple_ptr_holder.hpp

```diff
@@ -30,22 +30,31 @@ namespace gridtools {
             template <class T>
             struct simple_ptr_holder {
                 T m_val;
+
+#if defined(__cpp_deduction_guides) and __cpp_deduction_guides >= 201907
+// CTAD for aggregates works
+#else
+                // Enables CTAD in C++17.
+                simple_ptr_holder() = default;
+                GT_TARGET GT_FORCE_INLINE constexpr simple_ptr_holder(T const &ptr) : m_val{ptr} {}
+#endif
                 GT_TARGET GT_FORCE_INLINE constexpr T const &operator()() const { return m_val; }
             };
 
             template <class T>
-            constexpr simple_ptr_holder<T> make_simple_ptr_holder(T const &ptr) {
+            [[deprecated("use simple_ptr_holder class template argument deduction")]] constexpr simple_ptr_holder<T>
```

Comment:
Are you sure you want to keep the maker? I thought it is not the part of the public API.

Reply:
I'll keep it for now.